tools/c7n_gcp/c7n_gcp/resources/loadbalancer.py

```diff
@@ -435,3 +438,129 @@ def get(client, resource_info):
             return client.execute_command('get', {
                 'project': resource_info['project_id'],
                 'address': resource_info['resourceName'].rsplit('/', 1)[-1]})
+
+
+@resources.register('loadbalancer-security-policy')
+class LoadBalancingSecurityPolicy(QueryResourceManager):
+    """GCP resource: https://cloud.google.com/compute/docs/reference/rest/v1/securityPolicies"""
+    class resource_type(TypeInfo):
+        service = 'compute'
+        version = 'v1'
+        component = 'securityPolicies'
+        id = 'name'
+
+        @staticmethod
+        def get(client, resource_info):
+            return client.execute_command(
+                'get', {'project': resource_info['project_id'],
+                        'securityPolicy': resource_info['policy_name']})
+
+
+@LoadBalancingSecurityPolicy.action_registry.register('delete')
+class LoadBalancingSecurityPolicyDelete(MethodAction):
+    """
+    `Deletes <https://cloud.google.com/compute/docs/reference/rest/v1/securityPolicies/delete>`_
+    a security policy
+
+    :Example:
+
+    .. code-block:: yaml
+
+        policies:
+          - name: security-policy-delete
+            description: Deletes a security policy
+            resource: gcp.loadbalancer-security-policy
+            filters:
+              - type: value
+                key: name
+                value: test-policy
+            actions:
+              - delete
+    """
+
+    schema = type_schema('delete')
+    method_spec = {'op': 'delete'}
+    path_param_re = re.compile('.*?/projects/(.*?)/global/securityPolicies/(.*)')
+
+    def get_resource_params(self, m, r):
+        project, policy = self.path_param_re.match(r['selfLink']).groups()
+        return {'project': project, 'securityPolicy': policy}
+
+
+@LoadBalancingSecurityPolicy.action_registry.register('add')
+class LoadBalancingSecurityPolicyAddRule(MethodAction):
```

Comment:
what happens if this is used to add a duplicate rule

Reply:
It's not allowed to have more than one rule with the same priority so this will result into a http error (400). Seems that the details are swallowed by Custodian and there are no useful information in the traceback for a user. I will add the check and corresponding exception with details in case there are duplicates.
p.s. this still won't help in case something is added while custodian executes the policy

Reply:
@kapilt anything else here?

Reply:
@kapilt done.

Reply:
how is a user supposed to insert a priority value here apriori across a fleet of load balancer policies and projects that is correct and unique.. ie how is anyone supposed to use this at scale without getting runtime exceptions.

Reply:
@kapilt good point. I will delete the add rule action. Wrt resource. I would keep it because there is another useful action - [remove rule](https://cloud.google.com/compute/docs/reference/rest/v1/securityPolicies/removeRule). Use case for that: delete those allow-rules which include ip ranges being added to the blacklist. This action wasn't implemented yet because it requires a little bit more efforts. It will be added in the future I hope.

Reply:
@kapilt done

Reply:
Hello @kapilt Something else here?